twostars_2/twostars_m.f90

```diff
@@ -398,6 +404,119 @@ subroutine insolation2(mass,teff,rs,semia,e1,man,spin,t,nbands,bands,bndflux,dec
  return
 end subroutine
 
+!********************************************************************************************************************************************
+subroutine insolation3(spectra,mass,rs,semia,e1,man,spin,t,nbands,bands,bndflux,decl,phi,d2,dr1,dr2,weights) !ADZ CHANGE 20190804
+!-----------------------------------------------------------------------------------------------------------------------------------------
+! Determine the top-of the atmosphere insolation a circumbinary (P-type) planet receives in 
+! specified wave-length bands
+!
+! This routine lets you specify mass, solar radii, and spectrum for each star directly.
+! 
+!*)planetary and stellar dynamics are solved analytically using relations from Georgakarakos & Eggl 2014
+!
+! written by Siegfried Eggl 20140609
+!
+! modified by Anna Zuckerman 20190815
+!
+! dependencies: pconst, pdist, getspin, planetorient, intplanckwn, transitfactor
+!-----------------------------------------------------------------------------------------------------------------------------------------
+! Input
+!
+! real::
+! mass(1:3)      		... [Msun] 	m0, m1, m2: 	masses of the primary star, secondary star, and the planet
+! rs(1:2)        		... [m] 			stellar (mean) radii                                         !ADZ 20190815 REMOVE teff 
+! semia(1:2)     		... [au]   	a1, a2: 	semimajor axes of the binary (inner) and planetary (outer) orbit. 0<a1<a2  
+! e1             		... [] 				eccentricity of binary star's orbit  0 <= e1 < 1 / the planet will always start on a circular orbit      
+! man(1:2)       		... [rad]	ma10, ma20:	inital positions of the binary/planet on its orbit (initial mean longitudes), range [0,360[ deg
+! spin(1:4)      		... [mixed] 	(1) obl [deg]	obliquity (ref: z-axis = normal to the orbital plane), range [0,180[ deg
+! 						(2) pre [deg]	angle of precession (ref: x-axis = along initial pericenter of binary), range [0,360[ deg
+! 						(3) pp  [D]     rotation period of the planet
+!						(4) eta0 [deg]  initial hour angle of zero-meridian of planet wrt x-axis,   range [0,360[ deg
+! t		 		... [D]				current time in (Gaussian) days 
+! bands(1:2,1:nbands,1:2)       ... [cm^-1]			wave numbers of insolation band borders (first index: star, second index: band number, third index: lower and upper band limit wave numbers)
+! spectra                       ... [W m^-2 sr^-1] (1)star 1    spectra of star 1          !ADZ ADD 20190729
```

Comment:
After uploading this, I realized these units should be in photons m^-2 s^-1 A^-1 to match the example input spectra files. However, the units are somewhat arbitrary as they will be whatever units the spectra are given in in the input files, and they won't effect the calculations